src/js/helpers/nodesHelper.js

```diff
@@ -14,65 +14,81 @@
  * limitations under the License.
  */
 
-export const knotxNodes = () => {
-  const rootElem = document.body;
-  const COMMENTS = [];
-  const COMMENT_NODE_CODE = 8;
-  const HTML_NODE_CODE = 1;
-  // Fourth argument, which is actually obsolete according to the DOM4 standard, is required in IE 11
-  const iterator = document.createNodeIterator(
-    rootElem,
-    NodeFilter.SHOW_ALL,
-  );
-
-  let curNode;
-  let isBetweenComments = false;
-  let curComment = '';
-  // eslint-disable-next-line no-cond-assign
-  while ((curNode = iterator.nextNode())) {
-    if (
-      curNode.nodeType === COMMENT_NODE_CODE
-          && curNode.data.indexOf('data-knotx-id') !== -1
-    ) {
-      isBetweenComments = !isBetweenComments;
-      curComment = isBetweenComments ? curNode.data.match(/"([^']+)"/)[1] : '';
+const isFragmentBoundary = (node) => node.nodeType === Node.COMMENT_NODE && node.data.includes('data-knotx-id');
+
+const isCorrectFragmentNode = (node) => node.nodeType === Node.ELEMENT_NODE
+  || (node.nodeType === Node.TEXT_NODE && node.textContent.trim());
+
+const getKnotxId = (node) => node.data.match(/data-knotx-id="(.*)"/)[1];
+
+const isDebugDataNode = (node, fragmentId) => node && node.tagName === 'SCRIPT' && node.dataset.knotxId === fragmentId;
```

Comment:
Do these functions have their own tests?

Reply:
They don't. Should they? Afaik they would need to be exported for the purposes of testing only and i think current tests cover use cases well enough

Reply:
Exporting just for testing purpose is fine